rfcs/009 - Tree RFC.md

```diff
@@ -0,0 +1,110 @@
+# 009 - Tree RFC
+
+## Current Status
+
+### Proposed
+
+2020-09-25
+
+### Accepted
+
+YYYY-MM-DD
+
+#### Approvers
+
+- Luis de Pombo <luis@alantechnologies.com>
+
+### Implementation
+
+- [ ] Implemented: [One or more PRs](https://github.com/alantech/alan/some-pr-link-here) YYYY-MM-DD
+- [ ] Revoked/Superceded by: [RFC ###](./000 - RFC Template.md) YYYY-MM-DD
+
+## Author
+
+- David Ellis <david@alantechnologies.com>
+
+## Summary
+
+The ability to define and reason about data that has a tree-like structure is necessary to handle a host of different problems, from parsers and serializers, to manipulation of an HTML DOM or JSON object.
+
+This is an integral piece of any language. Most choose to allow recursive data types and then deal with the complexities around infinitely-defined types (one way or the other). For Alan, we instead define a new Tree type to handle this and provide a collection of tools to work with it.
```

Comment:
I was hoping the rfc would include some code samples into what it would look like to code/solve these kinds of  problem statements with `Tree`. That seems more relevant that the short hard notation which takes on a considerable amount of the rfc.

Reply:
Added :)